### ISSUE TITLE
Allow For Link Optimization if Hierarchy contains Bidirectional/Wikilink

### ISSUE THREAD

**Issue by shorty25h0r7** [dendronhq/dendron#708]
## Please Select if your Request is Either something new or an Enhancement ##

- [ ] Enhancement of an existing Feature.
- [x] Request of a new feature.

## Please select the area your request applys to. (Multiple selections are Possible) ##

- [ ] Onboard. Initial vault setup and import/export pods 
- [ ] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [x] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Is your feature request related to a problem? Please describe ##

I have found sometimes I will [[link]] to notes that already exist in the current note's ancestry, which seems a little redundant...

## Describe the solution you'd like ##

I'm not even sure what the best solution would be, but **a** solution would be to (optionally) have [[wikilinks]] automatically remove brackets if the note already exists in the note's ancestry. (There could be configs to toggle this or specify which subtrees to apply this feature to, etc.)

## Additional context ##

A feature that should probably be added before this is to be able to jump to any note in the current note's ancestry (from say a breadcrumb at the top of the note).

**Comment by kpathakota** [comment]
Yeah, this is pretty clear. I think a breadcrumb to allow for easy ancestral jumping is a great idea!

**Comment by aleksey-rowan** [comment]
Hey, @manunamz,

> A feature that should probably be added before this is to be able to jump to any note in the current note's ancestry (from say a breadcrumb at the top of the note).

You can move up the hierarchy with `Ctrl + Shift + Up arrow` and you can move sort of down to the original note using `Alt + Left arrow` (which goes back in history).

> I have found sometimes I will [[link]] to notes that already exist in the current note's ancestry, which seems a little redundant...

I'm curious, why do you say it's redundant? Is there a specific use case?

I use the bullet journal method and all my daily notes `yyyy.mm.dd` have a link pointing to the current month's note `yyyy.mm` which would fall under your definition. I can't think of other examples of the top of my head, but simply wondering what you've been finding yourself linking to.
